ml-agents/mlagents/trainers/tests/test_demo_loader.py

```diff
@@ -27,3 +32,11 @@ def test_load_demo_dir():
 
     _, demo_buffer = demo_to_buffer(path_prefix + "/test_demo_dir", 1)
     assert len(demo_buffer["actions"]) == total_expected - 1
+
+
+def test_nonexisting_paths():
+    path_prefix = os.path.dirname(os.path.abspath(__file__))
+    with pytest.raises(FileNotFoundError):
+        get_demo_files(os.path.join(path_prefix, "idontexistfile.demo"))
+    with pytest.raises(FileNotFoundError):
+        get_demo_files(os.path.join(path_prefix, "idontexistdirectory"))
```

Comment:
If you want to test some of the other logic that raises ValueError (not a .demo file, or no demo files in the directory), you can use the python tempfile library (https://docs.python.org/3/library/tempfile.html) to create directories and files that will automatically be cleaned up. So something like

```python
with tempfile.TemporaryDirectory() as tmpdirname:
    # get_demo_files(tmpdirname) should raise a ValueError
    with open(tmpdirname + "/mydemo.notademo", 'w') as f:
        f.write("I'm not a demo")
    # get_demo_files(tmpdirname) should raise a ValueError
    with open(tmpdirname + "/mydemo.demo", 'w') as f:
        f.write("I'm a demo file.")
    # get_demo_files(tmpdirname) should return a list with mydemo.demo in it
```
(that's just a rough outline, you get to fill in the commented parts yourself)